### PR TITLE
Remove `flow-archive` dependency & `storage/remote` package

### PR DESCRIPTION
**Pull request by jordanschalm**
Closes #???

## Description

- Remove `flow-archive` dependency & `storage/remote` package. See discussion on [Discord](https://discord.com/channels/613813861610684416/1108968095982293002/1181316536615567501) and [related issue](https://github.com/onflow/flow-go/pull/5071#issuecomment-1836534173).
- Update `flow-core-contracts` version to `v0.15.0`

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by m-Peter** [comment]
The tests are failing due to an upgraded version of `RandomBeaconHistory` contract. The new API should be used in the tests instead:

```cadence
let sorAt1 = RandomBeaconHistory.sourceOfRandomness(blockHeight: 1)
let sorAt2 = RandomBeaconHistory.sourceOfRandomness(blockHeight: 2)
```

The above lines should replace these ones: https://github.com/onflow/flow-emulator/blob/master/emulator/random_source_history_test.go#L83-L84

**Review by zhangchiqing — APPROVED** [review]

**Review by bluesign — APPROVED** [review]

File: emulator/random_source_history_test.go [hunk 1]
```diff
@@ -80,8 +80,8 @@ func TestRandomSourceHistoryAtBlockHeight(t *testing.T) {
 
         access(all)
         fun main(): Bool {
-            let sorAt1 = RandomBeaconHistory.sourceOfRandomnessAtBlockHeight(blockHeight: 1)
-            let sorAt2 = RandomBeaconHistory.sourceOfRandomnessAtBlockHeight(blockHeight: 2)
+            let sorAt1 = RandomBeaconHistory.sourceOfRandomness(atBlockHeight: 1)
+			let sorAt2 = RandomBeaconHistory.sourceOfRandomness(atBlockHeight: 2)
```
Comment (turbolent) [inline emulator/random_source_history_test.go:84]: Mixed tabs/spaces. Use spaces like the everywhere else:

```suggestion
            let sorAt1 = RandomBeaconHistory.sourceOfRandomness(atBlockHeight: 1)
            let sorAt2 = RandomBeaconHistory.sourceOfRandomness(atBlockHeight: 2)
```

File: go.mod [hunk 2]
```diff
@@ -14,8 +14,7 @@ require (
 	github.com/improbable-eng/grpc-web v0.15.0
 	github.com/logrusorgru/aurora v2.0.3+incompatible
 	github.com/onflow/cadence v0.42.5
-	github.com/onflow/flow-archive v1.3.4-0.20230503192214-9e81e82d4dcc
-	github.com/onflow/flow-core-contracts/lib/go/templates v1.2.4-0.20231016154253-a00dbf7c061f
+	github.com/onflow/flow-core-contracts/lib/go/templates v0.15.0
```
Comment (turbolent) [inline go.mod:17]: How come this is a "downgrade" from a commit above `v1.2.4` to `v0.15.0`?

Comment (jordanschalm) [reply go.mod:17]: `v1.2.3` was a tag that was pushed out of semver order a few months ago, which confuses the Go tooling. See https://github.com/onflow/flow-core-contracts/pull/398, which is retracting the incorrectly tagged version, for more context.

**Review by koko1123 — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

**Pull request merged**